### PR TITLE
made changes to allow k2-tools to use latest supported version of k8s…

### PR DESCRIPTION
**Pull request by alejandroEsc**
…. We have gone through enough issues where server  version is latest, but for whatever reason we do not have parity here. I suggest this is a move towards that direction.

### CODE REVIEW

**Comment by davidewatson** [comment]
LGTM. The build failed for some reason though:

```
[0m[91m/bin/sh: RUN: not found
[0mThe command '/bin/sh -c wget https://storage.googleapis.com/kubernetes-release/release/${K8S_VERSION_1_4}/bin/linux/amd64/kubectl &&         chmod a+x kubectl &&         mv kubectl /opt/cnct/kubernetes/v1.4/bin && RUN     wget https://storage.googleapis.com/kubernetes-release/release/${K8S_VERSION_1_5}/bin/linux/amd64/kubectl &&         chmod a+x kubectl &&         mv kubectl /opt/cnct/kubernetes/v1.5/bin' returned a non-zero code: 127
```

For reference:

The [documented policy](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/versioning.md#supported-releases-and-component-skew)[1] suggests we may run into problems whether we use the latest or the oldest k2 supported version of kubectl. k2cli uses the latest though. As does the kops [Dockerfile](https://github.com/kubernetes/kops/blob/b00e3fc41ddc124ca13ef495c74c0c56dd39d825/docker/Dockerfile-light) uses the latest released version (?= latest kops supported version)

[1]

> Different components are expected to be compatible across different amounts of skew, all relative to the master version. Nodes may lag masters components by up to two minor versions but should be at a version no newer than the master; a client should be skewed no more than one minor version from the master, but may lead the master by up to one minor version. For example, **a v1.3 master** should work with v1.1, v1.2, and v1.3 nodes, and **should work with v1.2, v1.3, and v1.4 clients.**

**Comment by alejandroEsc** [comment]
I think that is generally true, regardless of the version we use. However, i rather keep in line with the defaults used by k2, and every other tool, including k2cli should follow. What motivates this is the 1.6 release and the 1.4 kubectl not being able to produce human readable output, and it has occurred already to a great number of people in the team each time one of us having to recall that it was a kubectl version issue.

File: Dockerfile [hunk 1]
```diff
@@ -98,19 +98,19 @@ RUN     mkdir -p /opt/cnct/kubernetes/v1.4/bin \
                  /opt/cnct/kubernetes/v1.5/bin \
                  /opt/cnct/kubernetes/v1.6/bin \
                  /etc/helm/plugins && \
-        ln -s /opt/cnct/kubernetes/$LATEST /opt/cnct/kubernetes/latest
+        ln -s /opt/cnct/kubernetes/${LATEST} /opt/cnct/kubernetes/latest
 
 # Kubectl
 RUN     wget https://storage.googleapis.com/kubernetes-release/release/${K8S_VERSION_1_4}/bin/linux/amd64/kubectl && \
         chmod a+x kubectl && \
         mv kubectl /opt/cnct/kubernetes/v1.4/bin && \
```
Comment (joejulian) [inline Dockerfile]: Need to remove `&& \`

Comment (alejandroEsc) [inline Dockerfile]: correct!

Comment (davidewatson) [inline Dockerfile]: ?

**Comment by joejulian** [comment]
Before I merge this... do we want to do the same with helm?

**Comment by alejandroEsc** [comment]
ahhh, probably should to be honest. Ok, ill go ahead and make that change here as well. any objections?

**Comment by alejandroEsc** [comment]
@davidewatson @joejulian ready to be merged, thanks!

**Comment by davidewatson** [comment]
Given our lack of integration testing, we need to be careful here. I'd like to let this sit for a minute to allow other interested parties to take a look. I _think_ this LGTM.

References:

The version of `helm` used might be more restrictive than `kubernetes` [1]. A few related tickets are https://github.com/kubernetes/helm/issues/1283, 
https://github.com/kubernetes/helm/issues/1523, and https://github.com/kubernetes/helm/issues/1459.

[1]

The helm [policy](https://github.com/kubernetes/helm/blob/master/docs/install_faq.md):

> The rules for version numbers are as follows:
>
> Pre-release versions are incompatible with everything else. Alpha.1 is incompatible with Alpha.2.
> Patch revisions are compatible: 1.2.3 is compatible with 1.2.4
> **Minor revisions are not compatible: 1.2.0 is not compatible with 1.3.0, though we may relax this constraint in the future.**
> **Major revisions are not compatible: 1.0.0 is not compatible with 2.0.0.**

**Comment by alejandroEsc** [comment]
@davidewatson sounds fair. please ping them if you know a person(s) in mind. thanks again!

**Comment by davidewatson** [comment]
I was hoping @coffeepac or @guineveresaenger could weigh in since they have done the most with rolling upgrade support. I think this change only affects developers using the k2-tool image for k2 development though, and k2cli is selective about which version it uses, so perhaps they have no opinion.

**Comment by coffeepac** [comment]
Please wait for me to review this.

**Pull request merged**